neat_cache/lib/neat_cache.dart

```diff
@@ -175,23 +175,31 @@ class _Entry<T, V> implements Entry<T> {
       final created = await create();
       if (created != null) {
         // Calling `set(null)` is equivalent to `purge()`, we can skip that here
-        await set(created, ttl);
+        if (ttl != null) {
+          await set(created, ttl);
+        } else {
+          await set(created);
```

Comment:
why can't we use `set(created, ttl)` unconditionally?

Reply:
Because when you set a `ttl` you can'tse it to `null`, you can choose not to set it, but not set it to `null`.

We can debate the wisdom of such design... I'm okay changing it :D

Reply:
I don't see much harm in that, especially in cases like this, where there could be many different fallback config parameter that the client will use in the presence of `null`.